main.go

```diff
@@ -28,6 +28,7 @@ var (
 	fileNamePtr    = flag.String("file", "", "")
 	formatPtr      = flag.String("format", "", "")
 	followPtr      = flag.Bool("follow", false, "")
+	sortPtr        = flag.String("sort", "name", "")
```

Comment:
What are your thoughts on changing this to `sort-pkg`, so it's obvious which table is being sorted? 

I could see a future world where someone requests `sort-test` (name, elapsed, etc.) or even `sort-bench` (name, CPU, memory, etc.).

Reply:
I feel like adding specifics for pkg,test,bench forces the user to worry more about the specific output they get and which particular type of output should be sorted in which way. It would be simpler to have the user specify one parameter they care about the most based on their use case of what they want to analyze & optimize.

That said, some sorting preferences may not make sense particular test data, like sorting by coverage won't make sense for benchmarks and the cpu/memory/allocs won't make sense for regular tests. Hmm 🤔 

Personally I'd be fine if I pass benchmark data into `tparse -sort cover` if it just fell back to sorting by name. Or if I pass regular test data into `tparse -sort cpu` if it also just fell back to sorting by name. The output table wouldn't have that column, so it would become clear pretty quickly that I just specified a nonsense output format.

Another option would be to support a slice of sorting functions, so you could pass any data into `tparse -sort cpu,cover,name` and then for benchmarks it'll sort by CPU, for packages it'll sort by coverage, then everything falls back to name as the tiebreak.

Reply:
@mfridman what do you think, do you prefer different sort parameters for packages/benchmarks/tests?

Reply:
Was noodling on this for a bit. I quite like the general direction of how this is composed. 

So okay merging once the comments are addressed.

There is an outstanding question of how to sort the _tests table_, but we can cross that bridge later. So far sorting that table by elapsed has been sufficient.

Reply:
The tests table can be output with `-all` (which is `-pass` and `-skip` combined) and doesn't have much to do with `-follow`. Although for very large projects, I feel this table is only relevant if you're interested in capturing the slowest N tests per package. E.g., `tparse -all -slow 5` means show the table tests and only display the slowest 5.

The `-follow` was requested quite a bit because folks wanted to follow the output of `go test` despite running it with `-json`.

Not sure I understand this bit, but feedback welcome 😄 

> I wouldn't mind dropping the TEST pkg/foo lines from the makefile in the CI as well, but that's probably more of an item for me.

Reply:
>> I wouldn't mind dropping the TEST pkg/foo lines from the makefile in the CI as well, but that's probably more of an item for me.
>
> Not sure I understand this bit, but feedback is welcome 😄

In CI when the tests are running, it would be nice that each time the testsuite starts testing a new package, it prints one line with the package. This way, if one test gets stuck then that test is the last test that you see in the test output. So it's easy to tell which test you should look at. If there is no output until the end when the table is printed, then it can be hard to tell which package the CI got stuck on.

Reply:
Maybe we can open a new issue to keep the conversation open, otherwise, this will likely be lost in this thread.

Curious, is the combination of `-all -follow` not working for you? Example:

```
go test -v -count=1 crypto -json | tparse -all -follow 
```

The `-follow` prints _all_ go test output as if you hadn't run with `-json`, like this:

```
go test -v -count=1 crypto 
```

Reply:
Yeah I think the problem with that is that for large testsuites it becomes _too_ noisy. It prints N lines per test in the codebase, which is much larger than the number of packages. Too much output then becomes difficult to parse through again. I like the summary-only format, but that sometimes makes it hard to see which test hit problems if one of the tests just hangs rather than completing.

I can open another issue to discuss this further, maybe I can try to formulate my thoughts a bit more cohesively there.

Reply:
That'll be great, I work(ed) with fairly large code bases with loads of output so I _think_ I understand where you're getting at. There is probably some middle ground between no output and all, but I just haven't been able to find the right balance so input appreciated.